test/Microsoft.Azure.Devices.Edge.Test.Common/Profiler.cs

```diff
@@ -3,31 +3,31 @@ namespace Microsoft.Azure.Devices.Edge.Test.Common
 {
     using System;
     using System.Diagnostics;
+    using System.Linq;
     using System.Threading.Tasks;
+    using Serilog;
 
     public class Profiler
     {
-        public static Task Run(string startMessage, Func<Task> func, string endMessage = "") => Run(
-            startMessage,
+        public static Task Run(Func<Task> func, string message, params object[] properties) => Run(
             async () =>
             {
                 await func();
                 return true;
             },
-            endMessage);
+            message,
+            properties);
 
-        public static async Task<T> Run<T>(string startMessage, Func<Task<T>> func, string endMessage = "")
+        public static async Task<T> Run<T>(Func<Task<T>> func, string message, params object[] properties)
         {
-            Console.Write(startMessage + (string.IsNullOrEmpty(endMessage) ? "..." : "\n"));
             var stopwatch = new Stopwatch();
             stopwatch.Start();
 
             T t = await func();
 
             stopwatch.Stop();
-            TimeSpan ts = stopwatch.Elapsed;
-            Console.WriteLine(
-                (string.IsNullOrEmpty(endMessage) ? "done" : endMessage) + $" [{ts}]");
+            var args = new object[] { $"{((double)stopwatch.ElapsedMilliseconds) / 1000, 9:+0.000s}" };
+            Log.Information($"[{{Elapsed}}] {message}", args.Concat(properties).ToArray());
```

Comment:
Functions that provide the `(string format, params[] args)` signature (`Console.WriteLine`, `Microsoft.Extensions.Logging.Log.LogInformation`, etc) treat the format string literally if `args.Length == 0`, so that `format` is allowed to contain `{}`. Based on this assumption, the caller might do `Profiler.Run(() => { }, "a {b} c");`

Here you always provide at least one property (via `args`), so Serilog can't do that even if it could. So what does Serilog do if it encounters a format string that has more `{}` than the number of properties? If it just prints the unmatched `{}` content literally then it's fine, but if it throws an exception then it'll be a problem.

Reply:
I confirmed that Serilog just prints the unmatched content. So:
```
Profiler.Run(() => {}, "a {b} c");
```
prints:
```
[15:01:40 INF] [+  0.001s] a {b} c
```

Reply:
I'll add some unit tests to cover Profiler.

Reply:
Scratch that...Profiler is currently in the `/test` directory. To add tests I'd want to create `src` and `test` directories and move all files the files; seems excessive at this point.